addons/stock/models/stock_lot.py

```diff
@@ -213,6 +213,28 @@ def _search_product_qty(self, operator, value):
             return ['|', ('id', 'in', ids), ('id', 'not in', lot_ids_w_qty)]
         return [('id', 'in', ids)]
 
+    def _search_last_delivery_partner_id(self, operator, value):
+        lot_ids = self.env['stock.lot'].search([]).filtered(lambda line: line.last_delivery_partner_id)
```

Comment:
It will be super ineffective on large database with millions of lots. You could at least try to limit the set of lots